packages/ic-management/src/ic-management.canister.spec.ts

```diff
@@ -361,4 +361,50 @@ describe("ICManagementCanister", () => {
       expect(call).rejects.toThrowError(Error);
     });
   });
+
+  describe("provisionalCreateCanisterWithCycles", () => {
+    it("calls provisional_create_canister_with_cycles", async () => {
+      const response: ServiceResponse<
+        IcManagementService,
+        "provisional_create_canister_with_cycles"
+      > = {
+        canister_id: mockCanisterId,
+      };
+      const service = mock<IcManagementService>();
+      service.provisional_create_canister_with_cycles.mockResolvedValue(
+        response
+      );
+
+      const icManagement = await createICManagement(service);
+
+      await icManagement.provisionalCreateCanisterWithCycles();
+
+      expect(
+        service.provisional_create_canister_with_cycles
+      ).toHaveBeenCalledWith({
+        amount: [],
+        settings: [
+          {
+            controllers: [],
```

Comment:
not a must but, should we maybe also add a test with some parameters value? as we use `toNullable` to map those, maybe would be worth it? just an idea

Reply:
Yes that's a great point, I've updated the test now, thanks!